rest/oidc_api.go

```diff
@@ -106,6 +140,18 @@ func (h *handler) handleOIDCCallback() error {
 		return base.HTTPErrorf(http.StatusUnauthorized, "oidc callback received an error: %v", errorDescription)
 	}
 
+	stateCookie, err := h.rq.Cookie(stateCookieName)
+	if stateCookie == nil {
+		return ErrNoStateCookie
+	}
+	if err != nil {
+		return ErrReadStateCookie
+	}
```

Comment:
Idiomatically, error checking comes first - but I'd imagine it's this way around because rq.Cookie returns ErrNoCookie?

We should log a warning if we get a non-ErrNoCookie error, we don't expect it to happen given the current http.Request.Cookie implementation.

```suggestion
	if err != http.ErrNoCookie {
		base.Warnf("Unexpected error returned from request.Cookie(%s): %v", stateCookieName, err)
		return ErrReadStateCookie
	}
	if stateCookie == nil {
		return ErrNoStateCookie
	}
```

Reply:
Added changes to log a warning if we get a non-ErrNoCookie error.